src/goban.rs

```diff
@@ -68,7 +79,40 @@ impl Goban
 	}
 }
 
-#[cfg(test)]
+impl Eq for Goban {}
+
+impl PartialEq for Goban {
+	fn eq(&self, other: &Self) -> bool {
+		self.fscore == other.fscore
+	}
+}
+
+impl Ord for Goban {
+	fn cmp(&self, other: &Self) -> Ordering {
+		self.fscore.cmp(&other.fscore)
+	}
+}
+
+impl PartialOrd for Goban {
+	fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+		Some(self.cmp(other))
+	}
+}
+
+impl fmt::Display for Goban
+{
+	fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+		write!(f, "{}", self.player | self.enemy)
+	}
+}
+
+impl Hash for Goban {
+	fn hash<H: Hasher>(&self, state: &mut H) {
+		(self.player | self.enemy).hash(state)
+	}
+}
```

Comment:
I don't think this is the behavior we want, don't we rather want to use the value of `fscore` here?